receiver/prometheusreceiver/internal/transaction.go

```diff
@@ -223,7 +223,7 @@ func (t *transaction) detectAndStoreNativeHistogramStaleness(atMs int64, key *re
 
 // getOrCreateMetricFamily returns the metric family for the given metric name and scope,
 // and true if an existing family was found.
-func (t *transaction) getOrCreateMetricFamily(key resourceKey, scope scopeID, mn string) *metricFamily {
+func (t *transaction) getOrCreateMetricFamily(key resourceKey, scope scopeID, mn string, schema int32) *metricFamily {
```

Comment:
Something feels off with this strategy of adding a schema as an argument 🤔 

What if we took a similar approach to the one we use when detecting a NativeHistogram? When we call AppendHistogram, we toggle a flag in the `transaction` object called `addingNativeHistogram`. Couldn't we add the flag there, too?